server/src/processor.rs

```diff
@@ -0,0 +1,245 @@
+                }
+                None => {
+                    self.current_settled
+                        .insert(signal.account.clone(), signal.clone());
+                    self.settled_signals.push(signal);
+                }
+            },
+            SignalStatus::Unsettled => self.unsettled_signals.push(signal),
+            SignalStatus::Invalid => self.invalid_signals.push(signal),
+        }
+    }
+
+    fn get_stats_for(signals: Vec<Signal>, key: &str) -> SignalStats {
+        let mut stats: SignalStats = Default::default();
+        for signal in signals.iter() {
+            let delegated_balance = signal
+                .delegations
+                .delegated_balance
+                .parse::<f32>()
+                .unwrap_or(0.00);
+
+            if signal.memo.to_lowercase() == key.to_lowercase() {
+                stats.yes += delegated_balance;
+            }
+
+            if signal.memo.to_lowercase() == format!("no {}", key.to_lowercase()) {
+                stats.no += delegated_balance;
+            }
+        }
+
+        stats
+    }
+
+    pub fn settled_stats(&self) -> SignalStats {
+        let signals = self.current_settled.clone()
+            .into_iter()
+            .map(|(_, v)| v)
+            .collect::<Vec<Signal>>();
+
+        Self::get_stats_for(signals, &self.key)
+    }
+
+    pub fn stats(&self) -> SignalStats {
+        let mut total_stats: SignalStats = Default::default();
+
+        for (_account, signals) in self.signallers_cache.iter() {
+            let mut signals = signals.clone();
+            signals.sort_by(|x, y| x.height.cmp(&y.height));
+            if let Some(signal) = signals.get(0) {
+
+                if signal.signal_status != SignalStatus::Invalid {
+                    let delegated_balance = signal
+                        .delegations
+                        .delegated_balance
+                        .parse::<f32>()
+                        .unwrap_or(0.00);
+
+                    if signal.memo.to_lowercase() == self.key.to_lowercase() {
+                        total_stats.yes += delegated_balance;
+                    }
+
+                    if signal.memo.to_lowercase() == format!("no {}", &self.key.to_lowercase()) {
+                        total_stats.no += delegated_balance;
+                    }
+                }
+            }
+        }
+
+        total_stats
+    }
+
+    fn generate_response(self) -> ResponseEntity {
+        let mut stats_option = None;
+        let mut stats = self.stats();
+        let settled = self.current_settled
+            .into_iter()
+            .map(|(_, v)| v)
+            .collect::<Vec<Signal>>();
+
+        if stats.yes != 0. && stats.no != 0. {
+            stats_option = Some(stats);
+        }
+
+        ResponseEntity {
+            settled,
+            unsettled: self.unsettled_signals,
+            invalid: self.invalid_signals,
+            stats: stats_option
+        }
+    }
```

Comment:
You can match directly on the struct without shadowing `None` in `stats_option`:
```
fn generate_response(self) -> ResponseEntity {
        let stats = self.stats();
        let settled = self
            .current_settled
            .into_iter()
            .map(|(_, v)| v)
            .collect::<Vec<Signal>>();

        ResponseEntity {
            settled,
            unsettled: self.unsettled_signals,
            invalid: self.invalid_signals,
            stats: match stats.yes != 0. && stats.no != 0. {
                true => Some(stats),
                false => None,
            },
        }
    }